### PR TITLE
Create a publicly viewable edge object

### PR DESCRIPTION
**Pull request by sophaz**

### CODE REVIEW

File: classes/GViewerEdgeWriteObject.js [hunk 1]
```diff
@@ -3,35 +3,9 @@ const DB = require('../lib/db.js');
...
-  }
-
-  async _canCreate() {
-    return true;
-  }
-
-  async _canModify() {
-    return await this.canSee();
-  }
+class GViewerEdgeWriteObject extends GViewerEdgeObject {
```
Comment (sophaz) [inline classes/GViewerEdgeWriteObject.js]: I plan on deleting this in the next commit. Keeping it around just to not break things when this merges.

File: classes/GViewerEdgeObject.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  async _canModify() {
+    return await this.canSee();
+  }
+
+  _isViewerOrCreator() {
+    if (this.getViewer().getID() === this.getID()) {
+      return true;
+    }
+    var data = this.getData();
```
Comment (mnovati) [inline classes/GViewerEdgeObject.js]: var data = await this.getData()

File: classes/GViewerEdgeObject.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  async _canCreate() {
+    return true;
+  }
+
+  async _canModify() {
+    return await this.canSee();
+  }
+
+  _isViewerOrCreator() {
```
Comment (mnovati) [inline classes/GViewerEdgeObject.js]: async

File: classes/GViewerEdgeObject.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+const GObject = require('./GObject.js');
+const ReadAllViewer = require('./ReadAllViewer.js');
+
+class GViewerEdgeObject extends GObject {
+
+  async _canSee() {
+    if (this._isViewerOrCreator()) {
+      return true;
+    }
+    return this._viewerEdgeConnected();
```
Comment (mnovati) [inline classes/GViewerEdgeObject.js]: return await this....

File: classes/GViewerEdgeObject.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+const Constants = require('../lib/constants.js');
+const DB = require('../lib/db.js');
+
+const GObject = require('./GObject.js');
+const ReadAllViewer = require('./ReadAllViewer.js');
+
+class GViewerEdgeObject extends GObject {
+
+  async _canSee() {
+    if (this._isViewerOrCreator()) {
```
Comment (mnovati) [inline classes/GViewerEdgeObject.js]: var is_viewer_or_creator = await this._isVi....
if (is_viewer_or_creator) {

**Pull request merged**